README.md

```diff
@@ -15,29 +15,35 @@ For more detailed build instructions, see the Servo book under [Setting up your
 [Building for Android]: https://book.servo.org/hacking/building-for-android.html
 [Building for OpenHarmony]: https://book.servo.org/hacking/building-for-openharmony.html
 
+**NOTE**: In the below steps, please ensure you install at least version 3.9 of Python, preferably the latest version supported on your platform.
+Servo will use its own pinned version of Python (in the `.python-version` file), which it installs using [`uv`](https://docs.astral.sh/uv).
+The system version is required for now just to bootstrap the process of creating and activating the virtual environment.
+
 ### macOS
 
-- Download and install [`python`](https://www.python.org/downloads/macos/) (version 3.10 to 3.12), [Xcode](https://developer.apple.com/xcode/), and [`brew`](https://brew.sh/).
+- Download and install [`python`](https://www.python.org/downloads/macos/), [Xcode](https://developer.apple.com/xcode/), and [`brew`](https://brew.sh/).
+- Install `uv`: `curl -LsSf https://astral.sh/uv/install.sh | sh` 
 - Install `rustup`: `curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh`
 - Restart your shell to make sure `cargo` is available
 - Install the other dependencies: `./mach bootstrap`
 - Build servoshell: `./mach build`
 
 ### Linux
 
-- Install `curl` and `python` (version 3.10 to 3.12):
-  - Arch: `sudo pacman -S --needed curl python python-pip`
-  - Debian, Ubuntu: `sudo apt install curl python3-pip python3-venv python3-setuptools`
-  - Fedora: `sudo dnf install curl python3 python3-pip python3-devel`
-  - Gentoo: `sudo emerge net-misc/curl dev-python/pip`
+- Install `curl` and `python`:
+  - Arch: `sudo pacman -S --needed curl python`
+  - Debian, Ubuntu: `sudo apt install curl`
+  - Fedora: `sudo dnf install curl python3`
+  - Gentoo: `sudo emerge net-misc/curl`
+- Install `uv`: `curl -LsSf https://astral.sh/uv/install.sh | sh` 
 - Install `rustup`: `curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh`
 - Restart your shell to make sure `cargo` is available
 - Install the other dependencies: `./mach bootstrap`
 - Build servoshell: `./mach build`
 
 ### Windows
 
-- Download and install [`python`](https://www.python.org/downloads/windows/) (version 3.10 to 3.12), [`choco`](https://chocolatey.org/install#individual), and [`rustup`](https://win.rustup.rs/)
+- Download and install [`python`](https://www.python.org/downloads/windows/), [`uv`](https://docs.astral.sh/uv/getting-started/installation/#standalone-installer), [`choco`](https://chocolatey.org/install#individual), and [`rustup`](https://win.rustup.rs/)
```

Comment:
If `uv` is installed does Python also need to be installed or can the `uv` installation also handle the installation of Python?

Reply:
`uv` doesn't manage global python installations. We need a system python on the host so that `./mach` can be run as a regular command without first entering a virtual environment in which `uv's` python is in the PATH. My understanding is on Windows  and there maybe few linux distros don't have a default python3 installed globally - I remember `python` just opened Microsoft Store app in Windows last time I tried) - so this step is still needed.

In the future we could rewrite the `mach` entry point in shell script or find a different way to bootstrap to avoid system python entirely.

Reply:
Yes, I think that would be a good idea. At some point `mach` was both a shell script and a Python script.

Reply:
> Yes, I think that would be a good idea. At some point `mach` was both a shell script and a Python script.

Until https://github.com/servo/servo/pull/34485 lands it still is.

Reply:
> In the future we could rewrite the mach entry point in shell script or find a different way to bootstrap to avoid system python entirely.

We could potentially even use a rust script for this. Would avoid cross-platform issues with shell scripts (and python).

Reply:
Flying in to note that you can do this with `uv`:

```
#!/usr/bin/env -S uv run -s
# /// script
# dependencies = [
#   "requests",
# ]
# [tool.uv]
# exclude-newer = "2024-10-16T00:00:00Z"
# ///

...python code...
```

This allows you to run a file as a Python script using `uv`, within a venv with the needed deps installed. `exclude-newer` is a way to pin the dependencies since having a lockfile isn't supported in the preamble.

Reply:
I was looking into `uv run` as well. It does seem like it would simplify our bootstrap, but I think we'd still need a wrapper around it for NixOS as both `uv` and the Python binary come from the nix-shell environment. I was also concerned that `uv run` might be overkill for just bootstrapping as it runs in its own ephemeral virtual environment, but I'll have to do some more research there.